internal/manifests/mutate.go

```diff
@@ -0,0 +1,226 @@
+package manifests
+
+import (
+	"reflect"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	"github.com/imdario/mergo"
+	routev1 "github.com/openshift/api/route/v1"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// MutateFuncFor returns a mutate function based on the
+// existing resource's concrete type. It supports currently
+// only the following types or else panics:
+// - ConfigMap
+// - Service
+// - Deployment
+// - StatefulSet
+// - ServiceMonitor.
+func MutateFuncFor(existing, desired client.Object) controllerutil.MutateFn {
+	return func() error {
+		existingAnnotations := existing.GetAnnotations()
+		if err := mergeWithOverride(&existingAnnotations, desired.GetAnnotations()); err != nil {
+			return err
+		}
+		existing.SetAnnotations(existingAnnotations)
+
+		existingLabels := existing.GetLabels()
+		if err := mergeWithOverride(&existingLabels, desired.GetLabels()); err != nil {
+			return err
+		}
+		existing.SetLabels(existingLabels)
+
+		if ownerRefs := desired.GetOwnerReferences(); len(ownerRefs) > 0 {
+			existing.SetOwnerReferences(ownerRefs)
+		}
+
+		switch existing.(type) {
+		case *corev1.ConfigMap:
+			cm := existing.(*corev1.ConfigMap)
+			wantCm := desired.(*corev1.ConfigMap)
+			mutateConfigMap(cm, wantCm)
+
+		case *corev1.Service:
+			svc := existing.(*corev1.Service)
+			wantSvc := desired.(*corev1.Service)
+			return mutateService(svc, wantSvc)
```

Comment:
Just a question, why is `mutateService` returned and all the other mutate functions not?

Reply:
The other functions don't return error.